haystack/retriever/dense.py

```diff
@@ -186,13 +202,14 @@ def _prepare_model(self, encoder, saved_state, prefix):
         encoder.eval()
 
         # load weights from the model file
-        model_to_load = encoder.module if hasattr(encoder, 'module') else encoder
+        model_to_load = encoder.base_model.bert_model if hasattr(encoder, 'base_model') and hasattr(encoder.base_model,
```

Comment:
why do we need the prepare_model function? Cant we remove it because it seems super fb dpr code specific?

Reply:
I'll address this while refactoring the model loading